packages/core/src/configuration/Configuration.spec.ts

```diff
@@ -0,0 +1,146 @@
+            credentials: {
+              token: 'xxxxxxx.xxxx.xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx'
+            }
+          })
+      ).toThrow();
+    });
+
+    it('should throw an error if credentials or credential providers are not passed', () => {
+      expect(
+        () =>
+          new Configuration({
+            cluster: 'example.com'
+          })
+      ).toThrow();
+    });
+
+    it.each([
+      {
+        input: 'localhost',
+        expected: { bus: 'amqp://localhost:5672', api: 'http://localhost:8000' }
+      },
+      {
+        input: 'localhost:8080',
+        expected: { bus: 'amqp://localhost:5672', api: 'http://localhost:8000' }
+      },
+      {
+        input: 'http://localhost',
+        expected: { bus: 'amqp://localhost:5672', api: 'http://localhost:8000' }
+      },
+      {
+        input: 'http://localhost:8080',
+        expected: { bus: 'amqp://localhost:5672', api: 'http://localhost:8000' }
+      },
+      {
+        input: '127.0.0.1',
+        expected: { bus: 'amqp://127.0.0.1:5672', api: 'http://127.0.0.1:8000' }
+      },
+      {
+        input: '127.0.0.1:8080',
+        expected: { bus: 'amqp://127.0.0.1:5672', api: 'http://127.0.0.1:8000' }
+      },
+      {
+        input: 'http://127.0.0.1',
+        expected: { bus: 'amqp://127.0.0.1:5672', api: 'http://127.0.0.1:8000' }
+      },
+      {
+        input: 'http://127.0.0.1:8080',
+        expected: { bus: 'amqp://127.0.0.1:5672', api: 'http://127.0.0.1:8000' }
+      },
+      {
+        input: 'example.com',
+        expected: {
+          bus: 'amqps://amq.example.com:5672',
+          api: 'https://example.com'
+        }
+      },
+      {
+        input: 'example.com:443',
+        expected: {
+          bus: 'amqps://amq.example.com:5672',
+          api: 'https://example.com'
+        }
+      },
+      {
+        input: 'http://example.com',
+        expected: {
+          bus: 'amqps://amq.example.com:5672',
+          api: 'https://example.com'
+        }
+      },
+      {
+        input: 'http://example.com:443',
+        expected: {
+          bus: 'amqps://amq.example.com:5672',
+          api: 'https://example.com'
+        }
+      }
+    ])('should generate correct api and bus for $input', entry => {
+      const configuration = new Configuration({
+        cluster: entry.input,
+        credentials: {
+          token: 'xxxxxxx.xxxx.xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx'
+        }
+      });
+
+      expect(configuration.bus).toEqual(entry.expected.bus);
+      expect(configuration.api).toEqual(entry.expected.api);
+    });
+  });
+
+  describe('loadCredentials', () => {
+    it('should not throw if provider not defined', async () => {
+      const configuration = new Configuration({
+        cluster: 'app.neuralegion.com',
+        credentials: {
+          token: 'xxxxxxx.xxxx.xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx'
+        }
+      });
+      await expect(configuration.loadCredentials()).resolves.not.toThrow();
+    });
```

Comment:
Just noticed tath this test doesn't test anything) You should make sure that `credentials` are preserved